### ISSUE TITLE
Bug: NLog gets confused when using multiple config files.

### ISSUE THREAD

**Issue by sllewellyn13** [NLog/NLog#2521]
NLog gets confused when using multiple config files.

I have created App.config which contains App.Debug.cong and App.Release.config.
I have NLog.config in a separate file.
For my App.config I have to use transforms for security reasons.
When I try my new build, NLog fails utterly.

**NLog version**:   **4.4.12**

**Platform**: .Net 4.5  

**Current NLog config** (xml or C#, if relevant)

```xml
<?xml version="1.0" ?>
<nlog 
    xmlns="http://www.nlog-project.org/schemas/NLog.xsd" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    >
    <section name="nlog" type="NLog.Config.ConfigSectionHandler, NLog"/>
    <variable name="logDirectory" value="${basedir}\${shortdate}"  />
    <targets>
        <target name="logfile" xsi:type="File" fileName="bin\testlogs.log"  />
        <target name="console" xsi:type="Console" />
        <target name ="jsonlog" xsi:type="File" fileName = "testlogs.json" />
        <target xsi:type="File" name="file"  fileName="${logFileLocation}/${LogDay}.log" encoding="utf-8" layout="${longdate} ${uppercase:${level}} ${message}" />
    </targets>
  <rules>
    <logger name="*" minlevel="Trace" writeTo="logfile" />
    <logger name="*" minlevel="Trace" writeTo="console" />
  </rules>
  <include file="${basedir}/nlog.config" />
  </nlog>


```

In case of a BUG:

- What is the current result? TypeInitializationException:
```
SetUp failed for test fixture WitsStatusNunitTests.WotcTests
System.TypeInitializationException : The type initializer for 'WitsStatusNunitTests.BaseTest' threw an exception.
  ----> System.TypeInitializationException : The type initializer for 'NLog.Common.InternalLogger' threw an exception.
  ----> System.Configuration.ConfigurationErrorsException : Configuration system failed to initialize
  ----> System.Configuration.ConfigurationErrorsException : Unrecognized attribute 'xmlns:xdt'. Note that attribute names are case-sensitive. (D:\Users\sllewellyn\Source\Repos\DisplayWITS\WitsStatusNunitTests\WitsStatusNunitTests\bin\Debug\WitsStatusNunitTests.dll.config line 2)
   at WitsStatusNunitTests.BaseTest..ctor()
   at WitsStatusNunitTests.WotcTests..ctor()
--TypeInitializationException
```

```
NUnit Adapter 3.9.0.0: Test execution started
Running selected tests in D:\Users\sllewellyn\Source\Repos\DisplayWITS\WitsStatusNunitTests\WitsStatusNunitTests\bin\Debug\WitsStatusNunitTests.dll
NUnit3TestExecutor converted 162 of 162 NUnit test cases
SetUp failed for test fixture WitsStatusNunitTests.WotcTests
System.TypeInitializationException : The type initializer for 'WitsStatusNunitTests.BaseTest' threw an exception.
  ----> System.TypeInitializationException : The type initializer for 'NLog.Common.InternalLogger' threw an exception.
  ----> System.Configuration.ConfigurationErrorsException : Configuration system failed to initialize
  ----> System.Configuration.ConfigurationErrorsException : Unrecognized attribute 'xmlns:xdt'. Note that attribute names are case-sensitive. (D:\Users\sllewellyn\Source\Repos\DisplayWITS\WitsStatusNunitTests\WitsStatusNunitTests\bin\Debug\WitsStatusNunitTests.dll.config line 2)
   at WitsStatusNunitTests.BaseTest..ctor()
   at WitsStatusNunitTests.WotcTests..ctor()
--TypeInitializationException
   at NLog.Common.InternalLogger.Log(Exception ex, LogLevel level, String message)
   at NLog.Internal.ExceptionHelper.MustBeRethrown(Exception exception)
   at NLog.LogFactory.get_Configuration()
   at NLog.LogFactory.GetLogger(LoggerCacheKey cacheKey)
   at NLog.LogFactory.GetLogger(String name)
   at NLog.LogManager.GetLogger(String name)
   at WitsStatusNunitTests.BaseTest..cctor() in D:\Users\sllewellyn\Source\Repos\DisplayWITS\WitsStatusNunitTests\WitsStatusNunitTests\BaseTest.cs:line 32
--ConfigurationErrorsException
   at System.Configuration.ConfigurationManager.PrepareConfigSystem()
   at System.Configuration.ConfigurationManager.get_AppSettings()
   at NLog.Common.InternalLogger.GetSettingString(String configName, String envName)
   at NLog.Common.InternalLogger.GetSetting[T](String configName, String envName, T defaultValue)
   at NLog.Common.InternalLogger.Reset()
   at NLog.Common.InternalLogger..cctor()
--ConfigurationErrorsException
   at System.Configuration.ConfigurationSchemaErrors.ThrowIfErrors(Boolean ignoreLocal)
   at System.Configuration.BaseConfigurationRecord.ThrowIfParseErrors(ConfigurationSchemaErrors schemaErrors)
   at System.Configuration.BaseConfigurationRecord.ThrowIfInitErrors()
   at System.Configuration.ClientConfigurationSystem.EnsureInit(String configKey)
System.TypeInitializationException: The type initializer for 'NLog.Common.InternalLogger' threw an exception. ---> System.Configuration.ConfigurationErrorsException: Configuration system failed to initialize ---> System.Configuration.ConfigurationErrorsException: Unrecognized attribute 'xmlns:xdt'. Note that attribute names are case-sensitive. (D:\Users\sllewellyn\Source\Repos\DisplayWITS\WitsStatusNunitTests\WitsStatusNunitTests\bin\Debug\WitsStatusNunitTests.dll.config line 2)
   at System.Configuration.ConfigurationSchemaErrors.ThrowIfErrors(Boolean ignoreLocal)
   at System.Configuration.BaseConfigurationRecord.ThrowIfParseErrors(ConfigurationSchemaErrors schemaErrors)
   at System.Configuration.BaseConfigurationRecord.ThrowIfInitErrors()
   at System.Configuration.ClientConfigurationSystem.EnsureInit(String configKey)
   --- End of inner exception stack trace ---
   at System.Configuration.ConfigurationManager.PrepareConfigSystem()
   at System.Configuration.ConfigurationManager.get_AppSettings()
   at NLog.Common.InternalLogger.GetSettingString(String configName, String envName)
   at NLog.Common.InternalLogger.GetSetting[T](String configName, String envName, T defaultValue)
   at NLog.Common.InternalLogger.Reset()
   at NLog.Common.InternalLogger..cctor()
   --- End of inner exception stack trace ---
   at NLog.Common.InternalLogger.Error(Exception ex, String message)
   at NLog.LogFactory.OnLoggerShutdown(Object sender, EventArgs args)
   at NLog.Internal.Fakeables.AppDomainWrapper.OnDomainUnload(Object sender, EventArgs e)
Configuration system failed to initialize
   at NLog.Common.InternalLogger.Error(Exception ex, String message)
   at NLog.LogFactory.OnLoggerShutdown(Object sender, EventArgs args)
   at NLog.Internal.Fakeables.AppDomainWrapper.OnDomainUnload(Object sender, EventArgs e)
NUnit Adapter 3.9.0.0: Test execution complete
========== Run test finished: 1 run (0:00:02.2945872) ==========
```

**Comment by 304NotModified** [comment]
Could we see the app.config? Before and after transformation?

**Comment by sllewellyn13** [comment]
I'm unable to send that to you, Julian, because of security reasons.  -sulu

**Comment by 304NotModified** [comment]
Could you create a small repo then?

I can't guess what's in you app.config of course.

**Comment by sllewellyn13** [comment]
Here you go:
<?xml version="1.0"?>
<configuration>
 <configSections>
  
  <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77123419341234" requirePermission="false"/>
 </configSections>
 <startup>
  <supportedRuntime version="v4.5" sku=".NETFramework,Version=v4.5"/>
 </startup>
 <system.serviceModel>
  <bindings>
   <basicHttpBinding>
    <binding name="BasicHttpBinding_ISomeService"/>
   </basicHttpBinding>
   <wsHttpBinding>
    <binding name="wsbind">
     <security mode="Message">
      <transport clientCredentialType="Windows" proxyCredentialType="None"/>
      <message clientCredentialType="Windows" negotiateServiceCredential="true" algorithmSuite="Default" establishSecurityContext="true"/>
     </security>
    </binding>
   </wsHttpBinding>
  </bindings>
  <client>
   <endpoint address="http://somemachine/someapi/someservice.svc" binding="basicHttpBinding" bindingConfiguration="BasicHttpBinding_ISomeService" contract="SomeService.ISomeService" name="BasicHttpBinding_ISomeService"/>
  </client>
 </system.serviceModel>
 <appSettings xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
  <add key="appName" value="demoApp"/>
  <add key="UrlPath" value="abcd.SomeServiceAdmin/SomeService"/>
 </appSettings>
</configuration>

**Comment by 304NotModified** [comment]
I checked this, and we fixed this in 4.5.0 already by adding a try-catch

https://github.com/NLog/NLog/commit/169f9b8cd92283262c6585f8518adcb6e3e5afc5#diff-88b18910c9dcc05627dd9c753f15b829R486

![image](https://user-images.githubusercontent.com/5808377/38783076-a7ff4c02-40fd-11e8-85de-73f1734e3fb3.png)


We can't prevent the error from `ConfigurationManager.AppSettings`, but this will be catched

**Comment by sllewellyn13** [comment]
Thanks, you guys! I'll get the newer version.
